### DISCUSSION TITLE
valueTransform documentation about built-in transformers missing, is generator = casing the new suggestion?

### DISCUSSION THREAD

**Discussion by rklec** [dotnet/templating#7319] [Template authoring]
I have saved a reference to https://github.com/dotnet/templating/wiki/Available-Symbols-Generators as the go-to doc for this use case and have this `.template.config/template.json`:

```json
  "symbols": {
    "SomeName": {
      "isRequired": true,
      "type": "parameter",
      "replaces": "SomeName",
      "fileRename": "SomeName"
    },
    "SomeNameKebab": {
      "type": "derived",
      "valueSource": "SomeName",
      "valueTransform": "kebabCase",
      "description": "The same parameter as SomeName, but in kebab-case with hyphens, e.g. some-name.",
      "replaces": "some-name",
      "fileRename": "some-name"
    },
    "SomeNameCamel": {
      "type": "derived",
      "valueSource": "SomeName",
      "valueTransform": "firstLowerCaseInvariant",
      "description": "The same parameter as SomeName, but in camelCase, e.g. someName.",
      "replaces": "someName",
      "fileRename": "someName"
    },
  }
 ```
 
 This works due to built-in `valueTransform`'ers and IMHO this is great.
 
 Now I wanted to add a simple "all lowercase" transformation (and replacement symbol), and I thought of something like this:
 ```jsonc
    "SomeNameLower": {
      "type": "derived",
      "valueSource": "SomeName",
      "valueTransform": "lowerCase", // <-- This is my wild guess, I have not found a list of available valueTransform's
      "description": "The same parameter as SomeName, but in lowercase, e.g. somename.",
      "replaces": "somename",
      "fileRename": "somename"
    } 
```

The thing is I did not find many docs about the built-in generators of `valueTransform`.
Yes, https://github.com/dotnet/templating/blob/e393fd55ffb542ec07f6e29fc52ebca7479312c2/docs/Reference-for-template.json.md explains you can write your own transformator then, but where is the list of built-in ones?

Also I now see https://github.com/dotnet/templating/wiki/Available-Symbols-Generators recommends a generator instead of derived stuff for lowercasing. I guess it would apply like this:
```json
    "SomeNameLower": {
      "type": "generated",
      "generator": "casing",
      "parameters": {
        "source":"SomeName",
        "toLower": true
      },
      "description": "The same parameter as SomeName, but in lowercase, e.g. somename.",
      "replaces": "somename",
      "fileRename": "somename"
    }
```

I now wonder. Why this way then? Is this the new way?
Is `valueTransform` deprecated or so and should I use `"generator": "casing"` instead?

**PS:** A solution to this could also be to have a JSON schema, where I have autocompletion for `valueTransform`, so I can very easily get the available values.

**Comment by herbatnik** [comment]
~2.5 years and no response 😢 
I have the same problem right now and can't find any documentation for valueTransform options
